spring-data-solr/src/test/java/com/baeldung/solrjava/SolrJavaIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+package com.baeldung.solrjava;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.impl.XMLResponseParser;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrInputDocument;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SolrJavaIntegrationTest {
+
+    private HttpSolrClient solr;
+
+    @Before
+    public void setUp() throws Exception {
+
+        solr = new HttpSolrClient("http://localhost:8983/solr/bigboxstore");
+        solr.setParser(new XMLResponseParser());
+    }
+
+    @Test
+    public void givenAdd_thenVerifyAdded() throws SolrServerException, IOException {
```

Comment:
This test should probably be called "whenAdd_thenVerifyAdded".
Remember, in the "givenX_whenY_thenZ" convention, the "givenX" is optional, but the "whenY" should always be there.